components/dependency_injection/definitions.rst

```diff
@@ -69,12 +106,16 @@ or to get a single argument by its position::
     $definition->getArgument($index);
     // e.g. $definition->getArgument(0) for the first argument
 
-You can add a new argument to the end of the arguments array using::
+The argument can be a string, an array, a service parameter by using the
+``%parameter_name%`` syntax::
 
-    $definition->addArgument($argument);
+    $definition->addArgument('%kernel_debug%');
 
-The argument can be a string, an array, a service parameter by using ``%parameter_name%``
-or a service id by using::
+If the argument is another container service, don't use the ``get()`` method to
+get the actual service, because it won't be available when defining services.
```

Comment:
it may not be available at all (as it can be registered later). And it can also be defined in an incomplete way.
And finally, the container needs to know about how to define it later, so it needs the id